web/src/features/hfiCalculator/FormatTableAsCSV.ts

```diff
@@ -0,0 +1,252 @@
+                ? 'ND'
+                : station.station_props.elevation?.toString()
+            )
+            rowArray.push(station.station_props.fuel_type.abbrev)
+            rowArray.push(!isUndefined(daily) ? daily.status : 'ND')
+            rowArray.push(
+              !isUndefined(daily) && !isUndefined(daily.temperature)
+                ? daily.temperature.toFixed(DECIMAL_PLACES)
+                : 'ND'
+            )
+            rowArray.push(
+              !isUndefined(daily) && !isUndefined(daily.relative_humidity)
+                ? daily.relative_humidity.toFixed(DECIMAL_PLACES)
+                : 'ND'
+            )
+            rowArray.push(
+              !isUndefined(daily) && !isNull(daily.wind_direction)
+                ? daily.wind_direction.toString()
+                : 'ND'
+            )
+            rowArray.push(
+              !isUndefined(daily) && !isUndefined(daily.wind_speed)
+                ? daily.wind_speed.toFixed(DECIMAL_PLACES)
+                : 'ND'
+            )
+            rowArray.push(
+              !isUndefined(daily) && !isUndefined(daily.precipitation)
+                ? daily.precipitation.toFixed(DECIMAL_PLACES)
+                : 'ND'
+            )
+            rowArray.push(grassCureError ? 'ERROR' : printGrassCurePercentage(daily))
+            rowArray.push(!isUndefined(daily) ? daily.ffmc.toString() : 'ND')
+            rowArray.push(!isUndefined(daily) ? daily.dmc.toString() : 'ND')
+            rowArray.push(!isUndefined(daily) ? daily.dc.toString() : 'ND')
+            rowArray.push(!isUndefined(daily) ? daily.isi.toString() : 'ND')
+            rowArray.push(!isUndefined(daily) ? daily.bui.toString() : 'ND')
+            rowArray.push(!isUndefined(daily) ? daily.fwi.toString() : 'ND')
+            rowArray.push(
+              !isUndefined(daily) && !isNull(daily.danger_class)
+                ? daily.danger_class.toString()
+                : 'ND'
+            )
+            rowArray.push(
+              !isUndefined(daily) && !isUndefined(daily.rate_of_spread) && !grassCureError
+                ? daily.rate_of_spread.toFixed(DECIMAL_PLACES)
+                : 'ND'
+            )
+            rowArray.push(
+              !isUndefined(daily) && !isUndefined(daily.hfi) && !grassCureError
+                ? daily.hfi.toFixed(DECIMAL_PLACES)
+                : 'ND'
+            )
+            rowArray.push(
+              !isUndefined(daily) &&
+                !isUndefined(daily.sixty_minute_fire_size) &&
+                !grassCureError
+                ? daily.sixty_minute_fire_size.toFixed(DECIMAL_PLACES)
+                : 'ND'
+            )
+            rowArray.push(!isUndefined(daily) && !grassCureError ? daily.fire_type : 'ND')
+            rowArray.push(
+              !isUndefined(daily) && !grassCureError
+                ? daily.intensity_group.toString()
+                : 'ND'
+            )
+
+            rowsAsStrings.push(rowArray.toString())
```

Comment:
Uh oh - this isn't going to work - you can't just turn it into a string, for csv you have to escape things correctly.
e.g. - look at the output of:
`
[1, 2, 3, '4,5'].toString()
`
it returns the string: '1,2,3,4,5' - which is not what you want. The correct csv would be something like:
1,2,3,'4,5'

Reply:
It's not perfect but I don't see how this would ever actually break in production - HFI calc doesn't accept any user-provided values, and the values coming from the API are strictly numbers, so there won't be a `'` or anything characters like that in the values ever.

Reply:
I agree with @andrea-williams here. For our simple use case this works. If we start accepting user input / more complex CSV use cases then we should pull in a well-tested library that handles all the edge cases.

Reply:
I'm going to be obstinate here - I've seen this fail so many time. We have no control over what's going into the name field - it's very much outside of our control, and IS a user provided value. Someone is going around giving stations names.

It's really trivial if we use something like this: https://www.npmjs.com/package/csv-string

Reply:
True, I didn't think about station names.